src/org/zaproxy/zap/extension/pscanrules/CookieHttpOnlyScanner.java

```diff
@@ -66,32 +70,56 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 		if (cookies2 != null) {
 			iterator.addIterator(cookies2.iterator());
 		}
+		
+		List<String> ignoreList = new ArrayList<String>();
+		String ignoreConf = getModel().getOptionsParam().getConfig().
+				getString("rules.cookie.ignorelist");
+		if (ignoreConf != null && ignoreConf.length() > 0) {
+			for (String str : ignoreConf.split(",")) {
+				String strTrim = str.trim();
+				if (strTrim.length() > 0) {
+					ignoreList.add(strTrim);
+				}
+			}
+		}
 
 		while (iterator.hasNext()) {
 			String headerValue = (String) iterator.next();
 			if (!SetCookieUtils.hasAttribute(headerValue, HTTP_ONLY_COOKIE_ATTRIBUTE)) {
-				this.raiseAlert(msg, id, headerValue);
+				boolean ignoreCookie = false;
+				if (ignoreList.size() > 0) {
+					// Check its not one of the ones to ignore
+					for (String ignore : ignoreList) {
+						if (SetCookieUtils.getCookieName(headerValue).equals(ignore)) {
+							ignoreCookie = true;
+							break;
+						}
+					}
+				}
+				if (!ignoreCookie) {
```

Comment:
This if statement could use the contains (`SetCookieUtils.getCookieName(headerValue)`) method of cookie names collection (the previous if and for loop could be removed).